### ISSUE TITLE
Incorrect type assumption with int()

### ISSUE THREAD

**Issue by sbromberger** [tonyhffong/Lint.jl#79]
```
function lmatrix(g::Graph)
    n_v = nv(g)
    A = int(adjacency_matrix(g))   # note that this creates a sparse matrix of integers
    D = spdiagm(sum(A,2)[:])
    return D - A
end
```

yields

```
ERROR  sum: ErrorException("no method found for the specified argument types")
                                         Signature: (Int64,Int64)
```

Also curiously, it doesn't warn that I'm not using `n_v`.
